src/Workspaces/CoreTest/UtilityTest/FilePathUtilitiesTests.cs

```diff
@@ -99,5 +99,27 @@ public void GetRelativePath_WithBaseDirectoryMatchingIncompletePortionOfFullPath
 
             Assert.Equal(expected: @"..\Beta2\Gamma", actual: result);
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72043")]
+        public void GetRelativePath_CrashingWithTwoUp()
+        {
+            var baseDirectory = @"C:\Alpha\Beta";
+            var fullPath = @"C:\Gamma";
+
+            var result = PathUtilities.GetRelativePath(baseDirectory, fullPath);
+
+            Assert.Equal(expected: @"..\..\Gamma", actual: result);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72043")]
+        public void GetRelativePath_CrashingWithOneUp()
+        {
```

Comment:
Same feedback about Linux testing.

Reply:
Updated the tests, but made them conditional for Windows and Linux. 
Reason: Just flipping '\' with '/' for the test still returns '\' on Windows systems as this is the primary separator.
On Linux using a `C:\` pattern path feels odd to me and is in fact wrong as `:` is a path separator. I don't want the test to assert this, even the curremt implementation allows '\' and ':' to be used on Linux. But that'd be a different story.